### ISSUE TITLE
What changes I need to make to actually set the training process over multiple batches?

### ISSUE THREAD

**Issue by AAArpan** [explainingai-code/FasterRCNN-PyTorch#4]
Hey, I watched your video and loved it, can you tell me the changes need to make for training over multiple batches?

**Comment by explainingai-code** [comment]
Hello @AAArpan 
Glad you liked the video. For training over multiple batches there are three options.
1. Rather than using my faster rcnn model, use the official pytorch repo code. 
The file `tools/train_torchvision_frcnn.py` allows you to do that. I have talked about this in my next video on faster rcnn and you can use that to configure model accordingly and train for multiple batches. Adding a timestamped link for the part where I talk about fine tuning(https://youtu.be/YA7BqiUTCwM?t=3497) 
2. Use my faster rcnn model code only and train with batch size 1 but use gradient accumulation. Have talked about this [here](https://github.com/explainingai-code/FasterRCNN-PyTorch/issues/3#issuecomment-2395320704) 
2. Make changes to the current code to have the dataset return a batch of images all resized to same size. For this you would have to move below method to the dataset loader. 
    https://github.com/explainingai-code/FasterRCNN-PyTorch/blob/9c2f8eb1dc16d874b5427366ff46f7055e8d57d9/model/faster_rcnn.py#L762


    But this will resize each image such that min dimension > min_im_size(in config) and max dimension < max_im_size, so your 
    images will still be of different sizes.
    Then you can replicate what the pytorch official faster rcnn code does. So if your two images in batch after resizing above 
    are 800x600 and 600x1000, then you will create two empty images of 800x1000 and fill in pixels. In the video, I describe this resizing also,so you can go through it for reference(https://youtu.be/YA7BqiUTCwM?t=501)

Hope this helps.
